sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/expressions.scala

```diff
@@ -366,6 +363,16 @@ object BooleanSimplification extends Rule[LogicalPlan] with PredicateHelper {
             // ((c || ...) && (d || ...)) || a || b
             (common :+ And(ldiff.reduce(Or), rdiff.reduce(Or))).reduce(Or)
           }
+        } else {
+          // No common factors from disjunctive predicates, reduce common factor from conjunction
```

Comment:
nit: could you add an example case here like the comment above (` // (a || b || c || ...) && (a || b) => (a || b)`)?

Reply:
done.